### ISSUE TITLE
Enhance Error Message When Domain is Not Transferable

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#62506]
### What

When a domain is less than 60 days old, it cannot be transferred between registrars. When a user attempts to transfer-in a domain that is less than 60 days old, they are presented with a generic error message that does not say why they cannot transfer in the domain:

`The domain DOMAIN.com is not transferable.`

![Annotation on 2022-04-05 at 11-17-40](https://user-images.githubusercontent.com/30727666/161659028-e4291d12-25bd-4ea3-97bd-e3bbb3b9b024.png)

### Why

Without more information, the user is confused as to why they cannot transfer in the domain. They might abandon transferring in the domain completely, losing revenue for WordPress.com. 

Or, they might contact support to understand the issue. We want to reduce the number of users who need to contact support. Here's an example of the confusion this caused: #34966337-hc

### How

When we detect that a domain cannot be transferred-in because it's less than 60 days old, can we add that reason to the error message? I suggest updating the error message to:

>The domain DOMAIN.com cannot be transferred in because it is less than 60 days old. You may initiate the transfer on DATE. [Learn more about domain transfers.](https://wordpress.com/support/move-domain/incoming-domain-transfer/)

DATE is the 61st day after domain registration if this is information we have access to.

I have also linked to our support doc on the topic: https://wordpress.com/support/move-domain/incoming-domain-transfer/

It's also important that we let users know here that they don't have to wait 60 days to _use_ the domain, because they can map it using the name server method instead. On the screen, we mention this:

>If you would like to have your domain point to your WordPress.com site faster, consider [connecting your domain](https://wordpress.com/support/domains/map-existing-domain) first.

But in cases when a domain is <60 days old, we could consider more explicitly directing users down the domain connection flow to ensure they can move forward with their site.

**Comment by Greatdane** [comment]
I am not sure if there is any public tag that states a domain is less than 60 days old so it cannot be transferred, although I assume it is relatively easy to extrapolate by using whois and looking at the `Creation Date`.

**Comment by Robertght** [comment]
📌 REPRODUCTION RESULTS
- Tested on the platform - I couldn't replicate that exact flow anymore.

📌 FINDINGS/SCREENSHOTS/VIDEO
- While trying with a domain that was transferred out recently, I tried to see if that banner would trigger but it didn't and I made it up to the domain auth code step(2).
 
📌 ACTIONS
- Asking the team for further clarification.


📌 Message to the team
- @Automattic/nomado is this flow supposed to work like this or is currently working by design?

**Comment by ramynasr** [comment]
@Robertght Thanks for the ping!

## TL;DR:

We already detect when a domain is newly registered and show a message specifically for that state. Since this ticket was more than two years ago, I would suggest testing those conditions again and if they can't be reproduced, then we can probably close this one?

## More details:

Registrars *can* apply a transfer lock on a new registered or transferred-in domain. They have the option to lift that lock or not apply it at all, though. So, based on that:

* The transfer lock we detect might be related to other reasons, and not just the domain is newly registered.
* Some newly registered domains might not have that lock.
* If a domain is transferred-in recently, the lock can be applied, even though the creation date is years in the past. So, it is not straight-forward to detect the exact reason why the transfer lock is in place and the information might be misleading.

All that being said, we already detect when a domain is newly registered and show a message specifically for that state.

**Comment by Robertght** [comment]
Thank you @ramynasr! Are there any particular workflows or domains we could use to test this further? For example, I don't have a domain registered somewhere else, so I can't trigger these errors on my end.

Another thing I wanted to point is that I believe we could try and point the user direct to our documentation if we still have generic error messages. I imagine the above details(translated for our users) within the FAQ section could benefit everyone. What do you think?

**Comment by dsas** [comment]
If we've already separately done work that addresses this really old issue and we're confident about it, lets close it. There's plenty more bugs in the sea, and we can always reopen it if it turns out we need to.

If you want to verify it, I'd guess if you ask in a popular slack channel someone will have registered a domain elsewhere within the last two months.
